backend/app/services/chat/trigger/unified.py

```
@@ -222,6 +222,54 @@ async def build_execution_request(
+                    video_config["ratio"] = generate_params.ratio
+
+                if generate_params.duration:
+                    allowed_durations = capabilities.get("durations_sec") or []
+                    if (
+                        allowed_durations
+                        and generate_params.duration not in allowed_durations
+                    ):
+                        raise ValueError(
+                            f"Unsupported duration {generate_params.duration}s, "
+                            f"allowed: {allowed_durations}"
+                        )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)